js/parsers/paragraphparser.js

```diff
@@ -7,37 +7,66 @@ const pptelement_1 = require("airppt-models/pptelement");
  * Parse the paragraph elements
  */
 class ParagraphParser {
-    static extractParagraphElements(textElement) {
-        if (!textElement || !textElement["a:r"]) {
+    static extractParagraphElements(paragraphs) {
+        if (!paragraphs || paragraphs.length === 0) {
             return null;
         }
-        let pptTextElement = {
-            text: this.ConcatenateParagraphLines(textElement["a:r"]) || "",
-            textCharacterProperties: this.determineTextProperties(checkobj_1.CheckValidObject(textElement, '["a:r"][0]["a:rPr"][0]')),
-            paragraphProperties: this.determineParagraphProperties(textElement)
-        };
-        return pptTextElement;
+        return paragraphs.map((paragraph) => {
+            const textElements = paragraph["a:r"] || [];
+            const content = textElements.map((txtElement) => {
+                return {
+                    text: txtElement["a:t"] || "",
+                    textCharacterProperties: this.determineTextProperties(checkobj_1.CheckValidObject(txtElement, '["a:rPr"][0]'))
+                };
+            });
+            return {
+                content: content,
+                paragraphProperties: this.determineParagraphProperties(paragraph)
+            };
+        });
     }
     /**a:rPr */
     static determineTextProperties(textProperties) {
         if (!textProperties) {
             return null;
         }
-        let textPropertiesElement = {
+        const textPropertiesElement = {
             size: checkobj_1.CheckValidObject(textProperties, '["$"].sz') || 1200,
             fontAttributes: this.determineFontAttributes(textProperties["$"]),
             font: checkobj_1.CheckValidObject(textProperties, '["a:latin"][0]["$"]["typeface"]') || "Helvetica",
             fillColor: colorparser_1.default.getTextColors(textProperties) || "000000"
         };
         return textPropertiesElement;
     }
+    /** Parse for italics, bold, underline & strike through*/
+    static determineFontAttributes(attributesList) {
+        let attributesArray = [];
```

Comment:
```suggestion
        const attributesArray = [];
```
const array and objects can be mutated, they can't be reassigned.

Reply:
yea, i changed in lots of places already, missed this one.